RiotSwiftUI/Modules/Common/Util/MultilineTextField.swift

```diff
@@ -0,0 +1,213 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+@available(iOS 14.0, *)
+struct MultilineTextField: View {
+    
+    @Environment(\.theme) var theme: ThemeSwiftUI
+
+    private var placeholder: String
+
+    @Binding private var text: String
+    private var internalText: Binding<String> {
+        Binding<String>(get: { self.text } ) {
+            self.text = $0
+            self.showingPlaceholder = $0.isEmpty
+        }
+    }
+
+    @State private var dynamicHeight: CGFloat = 100
+    @State private var showingPlaceholder = false
+    @State private var isEditing = false
+
+    init(_ placeholder: String = "", text: Binding<String>) {
+        self.placeholder = placeholder
+        self._text = text
+        self._showingPlaceholder = State<Bool>(initialValue: self.text.isEmpty)
+    }
+    
+    private var textColor: Color {
+        if (theme.identifier == ThemeIdentifier.dark) {
+            return theme.colors.primaryContent
+        } else {
+            return theme.colors.primaryContent
+        }
+    }
+    
+    private var backgroundColor: Color {
+        return theme.colors.background
+    }
+    
+    private var placeholderColor: Color {
+        return theme.colors.tertiaryContent
+    }
+    
+    private var borderColor: Color {
+        if isEditing {
+            return theme.colors.accent
+        }
+        
+        return theme.colors.quarterlyContent
+    }
+    
+    private var borderWidth: CGFloat {
+        return isEditing ? 2.0 : 1.5
+    }
+    
+    var body: some View {
+        let rect = RoundedRectangle(cornerRadius: 8.0)
+        return UITextViewWrapper(text: self.internalText, calculatedHeight: $dynamicHeight, isEditing: $isEditing)
+            .frame(minHeight: dynamicHeight, maxHeight: dynamicHeight)
+            .padding(4.0)
+            .background(placeholderView, alignment: .topLeading)
+            .background(backgroundColor)
+            .clipShape(rect)
+            .overlay(rect.stroke(borderColor, lineWidth: borderWidth))
+            .introspectTextView { textView in
+                textView.textColor = UIColor(textColor)
+                textView.font = theme.fonts.uiFonts.callout
+            }
+    }
+
+    private var placeholderView: some View {
+        Group {
+            if showingPlaceholder {
+                Text(placeholder)
+                    .foregroundColor(placeholderColor)
+                    .font(theme.fonts.callout)
+                    .padding(.leading, 8.0)
+                    .padding(.top, 12.0)
+            }
+        }
+        .animation(.none)
+    }
```

Comment:
If you mark the variable with `@ViewBuilder` you can optionally return nothing and don't need the extra wrapping `Group`:
```suggestion
    @ViewBuilder
    private var placeholderView: some View {
            if showingPlaceholder {
                Text(placeholder)
                    .foregroundColor(placeholderColor)
                    .font(theme.fonts.callout)
                    .padding(.leading, 8.0)
                    .padding(.top, 12.0)
            }
    }
```